### PR TITLE
Fix execution time measurement

### PR DESCRIPTION
**Pull request by zpcore**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: examples/text_to_image/train_text_to_image_xla.py [hunk 1]
```diff
@@ -66,14 +66,13 @@ def run_optimizer(self):
         self.optimizer.step()
 
     def start_training(self):
-        times = []
-        last_time = time.time()
+        measure_start_step = 10
+        assert measure_start_step<self.args.max_train_steps
```
Comment (bhavya01) [inline examples/text_to_image/train_text_to_image_xla.py]: formatting nit: assert measure_start_step < self.args.max_train_steps

File: examples/text_to_image/train_text_to_image_xla.py [hunk 2]
```diff
@@ -66,14 +66,13 @@ def run_optimizer(self):
...
+        total_time = 0
         step = 0
+        last_time = time.time()
         while True:
-            if self.global_step >= self.args.max_train_steps:
-                xm.mark_step()
-                break
-            if step == 4 and PROFILE_DIR is not None:
+            if step ==  measure_start_step and PROFILE_DIR is not None:
                 xm.wait_device_ops()
```
Comment (bhavya01) [inline examples/text_to_image/train_text_to_image_xla.py:75]: Should we add `last_time = time.time()` after this xm.wait_device_ops()

File: examples/text_to_image/train_text_to_image_xla.py [hunk 3]
```diff
@@ -82,16 +81,13 @@ def start_training(self):
...
-            if step%5 == 0:
-                print(f'step: {step}, loss: {loss}')
-            last_time = time.time()
+            
             self.global_step += 1
             step += 1
-        print(f"Average step time: {sum(times)/len(times)}")
+            if step == self.args.max_train_steps:
+                xm.wait_device_ops()
+                total_time = time.time() - last_time
```
Comment (bhavya01) [inline examples/text_to_image/train_text_to_image_xla.py]: Should we add a break statement after this?

Comment (zpcore) [reply examples/text_to_image/train_text_to_image_xla.py]: Updated the condition to use for loop.

File: examples/text_to_image/train_text_to_image_xla.py [hunk 4]
```diff
@@ -66,33 +66,28 @@ def run_optimizer(self):
...
-                break
-            if step == 4 and PROFILE_DIR is not None:
+        dataloader_exception = False
+        measure_start_step = 10
+        assert measure_start_step < self.args.max_train_steps
+        total_time = 0
+        for step in range(0, self.args.max_train_steps):
+            if step ==  measure_start_step and PROFILE_DIR is not None:
                 xm.wait_device_ops()
+                last_time = time.time()
```
Comment (bhavya01) [inline examples/text_to_image/train_text_to_image_xla.py]: nit: Can we move this also after xp.trace_detached?

**Review by bhavya01 — APPROVED** [review]

**Pull request merged**